added base class for various actvities

sanyud24 opened the pull request:
cleaned the existing tests to inherit from the base test to follow DRY principle in the testing code.

![Screenshot from 2022-06-13 19-47-44](https://user-images.githubusercontent.com/57134307/173376442-f267f692-6963-4311-b30e-ec8341aa041f.png)

sanyud24 commented:
Hi @Lamouresparus @wbrunette , here are the updates for (Outreachy) week2.

wbrunette commented on services_app/src/androidTest/java/org/opendatakit/activites/LoginActivity/BaseLoginActivity.java:
```diff
@@ -0,0 +1,71 @@
+                    activity.getString(org.opendatakit.androidlibrary.R.string.default_sync_server_url)
+            );
+            assertThat(serverProperties).isNotNull();
+            serverProperties.put(CommonToolProperties.KEY_FIRST_LAUNCH,"true");
```
I am not sure why KEY_FIRST_LAUNCH set to true should be a clean for the test environment. We can discuss in our meeting.

sanyud24 commented:
Hi @wbrunette, did the required changes by setting up `KEY_FIRST_LAUNCH = false` for the default setup function and added a separate test when `KEY_FIRST_LAUNCH = true,` which opens a dialog box that configures server settings.

sanyud24 commented:
Hi @wbrunette, did the required changes 

- setting up `KEY_FIRST_LAUNCH = false` for the default setup function and added a separate test when `KEY_FIRST_LAUNCH = true`, which opens a dialog box that configures server settings for the **MainActivity.**

- As KEY_FIRST_LAUNCH is set to false for **AbsSyncBaseActivity** `props.setProperties(Collections.singletonMap(CommonToolProperties.KEY_FIRST_LAUNCH, Boolean.toString(false)));` , so removed the reductant code and updated the tests with `KEY_FIRST_LAUNCH = false`

- classes extending from AbsSyncBaseActivity : LoginActivity, SyncActivity, VerifyServerSettingActivity
